src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncStateActor.scala

```diff
@@ -0,0 +1,72 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.SupervisorStrategy.Escalate
+import akka.actor.{Actor, ActorLogging, OneForOneStrategy, SupervisorStrategy}
+import akka.pattern.pipe
+import io.iohk.ethereum.blockchain.sync.FastSyncController.SyncState
+import io.iohk.ethereum.blockchain.sync.FastSyncStateActor.GetStorage
+import io.iohk.ethereum.db.storage.FastSyncStateStorage
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Persists current state of fast sync to a storage. Can save only one state at a time.
+  * If during persisting new state is received then it will be saved immediately after current state
+  * was persisted.
+  * If during persisting more than one new state is received then only the last state will be kept in queue.
+  */
+class FastSyncStateActor extends Actor with ActorLogging {
+
+  override val supervisorStrategy: SupervisorStrategy = OneForOneStrategy(){ case _ => Escalate }
+
+  def receive: Receive = {
+    // after initialization send a valid Storage reference
+    case storage: FastSyncStateStorage => context become idle(storage)
```

Comment:
`Storage` is not static. Methods `.putSyncState` and `.purge` return new copy of a `Storage`.

It's not put into constructor in order to avoid storing reference to an outdated copy of `Storage`.